### DISCUSSION TITLE
Looking for User Defined Type (typeset -T) documentation/examples

### DISCUSSION THREAD

**Discussion by rwmills** [ksh93/ksh#730] [Q&A]
A couple of months ago I came across [Musings of an OS plumber](http://blog.fpmurphy.com/). The one posting that really caught my attention was [Using Types To Create Object Orientated Korn Shell 93 Scripts](https://blog.fpmurphy.com/2010/05/ksh93-using-types-to-create-object-orientated-scripts.html).

Ever since then I have been searching the Web looking for more information/examples of them. In all that time I've come to the conclusion that they are very rare because I have only found 3 examples and 3 documents (one of which is 'Type Variables' in the ksh93 man page).

If anybody here knows where I can find more on this feature then please can you pass it on to me. My plan is to write an article on this and post it here. I know of at least one person that would find it helpful.

**Comment by McDutchie** [comment]
I don't know of much either.

Have you found the file [`TYPES`](https://github.com/ksh93/ksh/blob/dev/src/cmd/ksh93/TYPES) that is included with the distribution? It's not much, but it's some more documentation. It could probably use a few edits.

**Reply by rwmills** [reply]
That's the first thing I found when I started searching.
